aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h

```diff
@@ -368,6 +379,17 @@ namespace impl {
     }
   };
 
+  template<class T, bool AllowDeprecatedTypes>
+  struct ivalue_to_arg<OptionalArrayRef<T>, AllowDeprecatedTypes> final {
+    // If an argument is optional<ArrayRef<T>>, convert the IValue to an
+    // optional<std::vector<T>> and pass that to the operator. OptionalArray<T>
+    // is basically a optional<std::vector<T>> but impliticly convertible to
+    // OptionalArrayRef<T>
+    static OptionalArrayRef<T> call(IValue& v) {
+      return ivalue_to_arg<OptionalArray<T>, AllowDeprecatedTypes>::call(v);
```

Comment:
At the moment, I'm getting errors with JITed functions that use `int[]?`. In these cases, when execution reaches the underlying `aten` function, the `OptionalIntArrayRef` contains garbage data. The root of the issue is right here.

After this `ivalue_to_arg<OptionalArray<T>, ...>::call(v)`, there is an implicit cast from `OptionalArray<T>` to `OptionalArrayRef<T>`. Below, in `aten/src/ATen/core/ivalue.h`, I added an overload for the cast operation of `OptionalArray<T>` so it could be cast to `OptionalArrayRef<T>`. Both the cast operation and this `ivalue_to_arg` class are implemented the same exact way as they were for `optional<ArrayRef<T>>`.

However, for some reason, the `OptionalArrayRef<T>` case acts slightly differently from the `optional<ArrayRef<T>>` case. Within the casting operator `OptionalArray::operator c10::OptionalArrayRef<T>()`, after the cast is complete and before the result is returned, the result looks correct. It can be printed, and the data within the `OptionalArrayRef<T>` is not garbage. But then after it's returned, and execution is back in `ivalue_to_arg<OptionalArrayRef<T>, ...>::call(IValue& v)`, the `OptionalArrayRef<T>` is no longer correct and it contains garbage.

I'm not yet sure why this happens. Maybe there's something wrong with how `IValue` interacts with the `OptionalArrayRef<T>`